dbms/src/Server/LocalServer.cpp

```diff
@@ -0,0 +1,399 @@
+#include "LocalServer.h"
+#include <Poco/Util/XMLConfiguration.h>
+#include <Poco/Util/HelpFormatter.h>
+#include <Poco/Util/OptionCallback.h>
+
+#include <DB/Databases/DatabaseOrdinary.h>
+
+#include <DB/Storages/System/StorageSystemNumbers.h>
+#include <DB/Storages/System/StorageSystemOne.h>
+
+#include <DB/Interpreters/Context.h>
+#include <DB/Interpreters/ProcessList.h>
+#include <DB/Interpreters/executeQuery.h>
+
+#include <DB/Common/Macros.h>
+#include <DB/Common/ConfigProcessor.h>
+#include <DB/Common/escapeForFileName.h>
+
+#include <DB/IO/ReadBufferFromString.h>
+#include <DB/IO/WriteBufferFromFileDescriptor.h>
+
+#include <DB/Parsers/parseQuery.h>
+#include <DB/Parsers/IAST.h>
+
+#include <common/ErrorHandlers.h>
+#include <common/ApplicationServerExt.h>
+
+
+namespace DB
+{
+
+void LocalServer::initialize(Poco::Util::Application & self)
+{
+	Poco::Util::Application::initialize(self);
+
+	/// Load config files if exists
+	if (config().has("config-file") || Poco::File("config.xml").exists())
+	{
+		ConfigurationPtr processed_config = ConfigProcessor(false, true).loadConfig(config().getString("config-file", "config.xml"));
+		config().add(processed_config.duplicate(), PRIO_DEFAULT, false);
+	}
+}
+
+void LocalServer::defineOptions(Poco::Util::OptionSet& _options)
+{
+	Poco::Util::Application::defineOptions (_options);
+
+	_options.addOption(
+		Poco::Util::Option ("config-file", "C", "Load configuration from a given file")
+			.required (false)
+			.repeatable (false)
+			.argument (" config.xml")
```

Comment:
Otherwise `--help` will print:
`-Cconfig.xml, --config-file=config.xml`
instead of
`-C config.xml, --config-file= config.xml`
`Cc` is ugly.